src/core/Ordering.ml

```diff
@@ -138,59 +156,49 @@ module KBO : ORD = struct
     balance.balance.(idx) <- n - 1
 
   let weight prec = function
-    | IOB.B _ -> W.int 1
-    | IOB.I s -> Prec.weight prec s
+    | Head.B _ -> W.int 1
+    | Head.I s -> Prec.weight prec s
+    | Head.V _ -> W.int 1  (* TODO: Maybe not a good value *)
 
   exception Has_lambda
 
-  (* TODO: ghc() that maps (in theory) variables to sets of symbols
-     their instances can have as head (for a symbol [f] it's just [{f}] itself).
-     Then [x > y] if [forall f∈ghc(x),y∈ghd(y), f>g] in precedence.
-  *)
-
-  (* FIXME: use [int ref HVar.Map] instead of an array, because there
-     might be variable collisions?
-     Check impact on perf *)
-
-  (** the KBO ordering itself. The implementation is borrowed from
-      the kbo_5 version of "things to know when implementing KBO".
-      It should be linear time. *)
+  (** Blanchette's higher-order KBO *)
   let rec kbo ~prec t1 t2 =
     let balance = mk_balance t1 t2 in
     (** variable balance, weight balance, t contains variable y. pos
         stands for positive (is t the left term) *)
     let rec balance_weight (wb:W.t) t y ~pos : W.t * bool =
       match T.view t with
-        | T.Var x ->
```

Comment:
you should use `ocp-indent` on this. It's a bit sad to have such big diffs only for whitespace :-)
I think the settings for ocp-indent should restore to the previous state.

edit: there is `make ocp-indent` if you have installed the tool via opam.